From 55fc37923f13c52d8bd18705da14e51aa9800ff6 Mon Sep 17 00:00:00 2001
From: Jeremias Cordoba <js.cordoba8321@gmail.com>
Date: Thu, 6 Nov 2025 14:51:57 -0800
Subject: [PATCH 1/1] Add support for torizon_boot_args in uEnv.txt

Save variable "torizon_boot_args" from previous OSTree deployment.
This allows the setting of custom boot args by external tooling,
in a way that OSTree can track.

Upstream-Status: Inappropriate [Torizon OS specific]

Related-to: TOR-4066

Signed-off-by: Jeremias Cordoba <jeremias.cordoba@toradex.com>
---
 src/libostree/ostree-bootloader-uboot.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/libostree/ostree-bootloader-uboot.c b/src/libostree/ostree-bootloader-uboot.c
index a566b997..361e8d9f 100644
--- a/src/libostree/ostree-bootloader-uboot.c
+++ b/src/libostree/ostree-bootloader-uboot.c
@@ -204,6 +204,17 @@ create_config_from_boot_loader_entries (OstreeBootloaderUboot *self, int bootver
               g_free (val);
             }
         }
+
+      if (i)
+        {
+          val = uboot_config_get ("torizon_boot_args", cancellable, error);
+          if (val)
+            {
+              g_ptr_array_add (new_lines,
+                               g_strdup_printf ("torizon_boot_args%s=%s", index_suffix, val));
+              g_free (val);
+            }
+        }
     }
 
   return TRUE;
-- 
2.30.2

